drivers/adc/ad7689/ad7689.c

```diff
@@ -0,0 +1,194 @@
+#include <stdlib.h>
+#include "ad7689.h"
+#include "util.h"
+#include "error.h"
+#include "print_log.h"
+#include "delay.h"
+
+const char *ad7689_device_name[] = {
```

Comment:
maybe add an attribute `name` to the device descriptor? (just a suggestion)
Some examples: 
https://github.com/analogdevicesinc/no-OS/blob/7c63e0dd4d5342106e0c7d33205ebdb5db5f5c2b/drivers/axi_core/jesd204/axi_jesd204_rx.h#L78-L79

https://github.com/analogdevicesinc/no-OS/blob/7c63e0dd4d5342106e0c7d33205ebdb5db5f5c2b/drivers/axi_core/jesd204/axi_adxcvr.h#L53

https://github.com/analogdevicesinc/no-OS/blob/7c63e0dd4d5342106e0c7d33205ebdb5db5f5c2b/drivers/axi_core/axi_dac_core/axi_dac_core.h#L58-L59

Reply:
yeah, this would be better